### PR TITLE
refactor/501 AdminView AdminUsers

### LINKED ISSUE #501: Refactor AdminView, AdminUsers and related components

### PR DESCRIPTION
**Pull request by te4-alexander-kjellberg**
## ⚙️ What issue did you implement or fix?
Fixes #501 

## 📜 Summary
- Refactored AdminView.vue
- Refactored AdminUsersView.vue

## 📝 Type of change

Please delete options that are not relevant.

- [x] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/views/AdminUsersView.vue [hunk 1]
```diff
@@ -33,57 +33,62 @@
...
-  },
-})
-export default class AdminUsersVIew extends Vue {
-  private UsersModule: VuexModule = UsersModule;
-  private RoleChecker: RoleChecker = RoleChecker;
-  private convertListToA: Function = convertListToA;
+export default ({
+  name: 'AdminUsersView',
+  setup(_: object, { root }: SetupContext): object {
+    const search: string = '';
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/views/AdminUsersView.vue:45]: Do this one need to be reactive? In other terms, does the search function still work?

File: frontend/src/views/Start.vue [hunk 2]
```diff
@@ -87,40 +87,73 @@
...
+//   // there is a relation between loans and users the constructor instantiates the class
+//   // by fetching all the information from the two modules
+
+//   public created(): void {
+//     if(UsersModule.currentUserID){
+//       UsersModule.fetchAll();
+//       LoansModule.fetchAll();
+//     }
+//   }
+// }
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/views/Start.vue:156]:
```suggestion
```

No need for that now

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
Looks promising!

**Review by Te4-Jacob-sandstrom — CHANGES_REQUESTED** [review]
Please make sure that all tests pass before requesting a review

**Comment by te4-alexander-kjellberg** [comment]
Anyone know how I can make these tests pass?

**Comment by te4-adrian-almetun-smeds** [comment]
Closing in favour of #523

**Pull request closed without merging**